### ISSUE TITLE
Snowflake objects created via terraform differs from "same" objects created via snowsight

### ISSUE THREAD

**Issue by Petruspop** [Snowflake-Labs/terraform-provider-snowflake#2356]
### Terraform CLI and Provider Versions

Terraform 1.6.6
Snowflake-Labs/snowflake 0.83.1

### Terraform Configuration

```terraform
#Set the provider
terraform {
  required_providers {
    snowflake = {
      source = "Snowflake-Labs/snowflake"
      version = "0.83.1"
    }
  }
}

# Configure the Snowflake Provider
# pwd as env var, set snowflake_password=your-password-here
provider "snowflake" {
  account = "RUTERAS-CDPPETER"
  user = "TMPTF"
  role = "ACCOUNTADMIN"
}

# Create DB with data retention set
resource "snowflake_database" "database" {
  name = "data-retention-db"
  data_retention_time_in_days = 28
  comment = "Database created in tf with data retention 28"
}

# Create schema without paramater for data retention -> should inherit
resource "snowflake_schema" "schema" {
  database = snowflake_database.database.name
  name = "tfschema"
  comment = "Schema created in tf, no data retention parameter used/set"
}

# Create table without paramater for data retention -> should inherit
resource "snowflake_table" "table" {
  database = snowflake_database.database.name
  schema = snowflake_schema.schema.name
  name = "tftable"
  comment = "Table created in tf, no value for data retention parameter"
  column {
    name = "id"
    type = "int"
  }
  column {
    name = "textfield"
    type = "text"
  }
}
```

### Expected Behavior

Data retention days on schema and table should be 28, as it should be inherited from database.

### Actual Behavior

Data retention days is defaulted to 1, both for schema and table.

### Steps to Reproduce

1. `terraform apply`
2. Check parameter on objects

### How much impact is this issue causing?

High

### Additional Information

Marked this as High as you expect you have time travel for 28 days, but in fact only have 1 day. So you risk losing data which is really dangerous.

The following script can be used in Snowsight to verify the difference. 

-- Verify TF defect
USE DATABASE "data-retention-db"; -- This was created in step 1 from tf provider

CREATE SCHEMA "uischema";

USE SCHEMA "uischema";

CREATE TABLE "uitable" (id int,textfield text) COMMENT ='Table created via snowsight, with no data retention parameter value';

SHOW PARAMETERS LIKE '%DATA_RETENTION%' IN DATABASE "data-retention-db";
--> 28 - as expected (from tf)

SHOW PARAMETERS LIKE '%DATA_RETENTION%' IN SCHEMA "uischema";
--> 28 - as expected, inherited from DB

SHOW PARAMETERS LIKE '%DATA_RETENTION%' IN SCHEMA "tfschema";
--> 1 - NOT as expected, should be the same as for the schema from Snowsight

SHOW PARAMETERS LIKE '%DATA_RETENTION%' IN TABLE "uitable";
--> 28 - as expected, inherited from DB->schema->table

USE SCHEMA "tfschema";
SHOW PARAMETERS LIKE '%DATA_RETENTION%' IN TABLE "tftable";
--> 1 - NOT as expected

**Comment by sfc-gh-asawicki** [comment]
Hey @Petruspop. Thanks for reporting the issue.

Based on the setup provided, the source of the error is probably using the default value from the resource schema. This should be a relatively easy fix; we will deliver it in a few days if my suspicions are confirmed.

**Comment by timgreni** [comment]
@sfc-gh-asawicki any update?

**Comment by sfc-gh-asawicki** [comment]
Hey @timgreni. No, the issue is still waiting on our end.

**Comment by sfc-gh-jcieslak** [comment]
@Petruspop @timgreni working on the issue; should be fixed next week

**Comment by sfc-gh-jcieslak** [comment]
Hey @Petruspop @timgreni. We have released the fix as part of [v0.87.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.87.0) release. Please follow the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/v0.87.0/MIGRATION_GUIDE.md#v0860--v0870) during the update. Please confirm that the issue is resolved in the newest version. Thanks!
